main.go

```diff
@@ -164,9 +160,36 @@ func proxy(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func main() {
-	flag.Parse()
+// livenessProbeHandler tests whether the http server has hung by attempting to return a simple request
+func livenessProbeHandler(w http.ResponseWriter, r *http.Request) {
+	return
+}
 
+// readinessProbeHandler attempts to get the metadata from each of a list of Google buckets provided as an argument via
+// --readiness-buckets. If it fails to retrieve metadata from ALL buckets provided the handler will return a 503
+// response. The default list of buckets provided contains the three public Google buckets (see
+// https://cloud.google.com/storage/docs/public-datasets).
+func readinessProbeHandler(w http.ResponseWriter, r *http.Request) {
+	var err error
+	for _, bucket := range strings.Split(*readinessBuckets, ",") {
+		_, err = client.Bucket(bucket).Attrs(ctx)
```

Comment:
should we parallelize this ?

what if there is a timeout to the first bucket and so it takes X number of seconds to return and fail everything even though only the first bucket was having issues ?

We should be able to do it by using `goroutines` https://tour.golang.org/concurrency/1

Reply:
Yeah, that's a great shout actually. Will have a look now.

Reply:
May have missed this - but what's the use-case for testing multiple buckets anyway?